### PR TITLE
Update C++ binding and function list

### PR DESCRIPTION
**Pull request by kleisauke**
Depends on https://github.com/libvips/pyvips/pull/141.

### CODE REVIEW

File: doc/gen-function-list.py [hunk 1]
```diff
@@ -53,15 +55,50 @@ def add_nickname(gtype, a, b):
...
 
-    # add 'missing' synonyms by hand
-    all_nicknames.append('crop')
-
     # make list unique and sort
     all_nicknames = list(set(all_nicknames))
     all_nicknames.sort()
 
+    # make dict with overloads
+    overloads = {
```
Comment (kleisauke) [inline doc/gen-function-list.py]: I don't think there's a way to automatically generate these overloads?

File: doc/gen-function-list.py [hunk 2]
```diff
@@ -53,15 +55,50 @@ def add_nickname(gtype, a, b):
...
+
+    overloads['relational_const'] = [o + '_const' for o in overloads['relational']] + ['relational_const1'] + \
+                                    [o + '_const1' for o in overloads['relational']]
+
     for nickname in all_nicknames:
-        gen_function(nickname)
+        result = gen_function(nickname)
+        if nickname in overloads:
+            result = result.replace('()', '(), ' + (', '.join('vips_{}()'.format(n) for n in overloads[nickname])))
+        print(result)
```
Comment (jcupitt) [inline doc/gen-function-list.py]: Very nice!

File: doc/gen-function-list.py (hunk 1 above)
Comment (kleisauke) [reply doc/gen-function-list.py]: (I copied and pasted them from the old function list, so maybe I forgot a few)

Comment (jcupitt) [reply doc/gen-function-list.py]: Ooop, didn't post my comment. Sorry, no, I don't think so, unfortunately.

**Pull request merged**